include/dba.php

```diff
@@ -915,13 +962,17 @@ static public function delete($table, $param, $in_commit = false, &$callstack =
 						logger(dba::replace_parameters($sql, $field_values), LOGGER_DATA);
 
 						if (!self::e($sql, $field_values)) {
-							self::p("ROLLBACK");
+							if (!self::$in_transaction) {
```

Comment:
I think this test for `$in_transaction` should be inside `rollback()`. Additionally, I would expect a positive check for `$in_transaction`, is the negative check intended?